### ISSUE TITLE
Support for validating custom types

### ISSUE THREAD

**Issue by kentandersen** [thedersen/backbone.validation#239]
Hi

There seems to be an issue when attempting to use custom types. I'm trying to validate a moment daterange object but none of the valdation methods is called. I've tried both custom vaidation rules, `_.extend(Backbone.Validation.validators, { ... })` and `"dateRange": function() {}`.

After some debugging, the flatten method is reveled  as the culprit.

```javascript
var flatten = function (obj, into, prefix) {
  into = into || {};
  prefix = prefix || '';

  _.each(obj, function(val, key) {
    if(obj.hasOwnProperty(key)) {
      if (val && typeof val === 'object' && !(
        val instanceof Array ||
        val instanceof Date ||
        val instanceof RegExp ||
        val instanceof Backbone.Model ||
        val instanceof Backbone.Collection)
      ) {
        flatten(val, into, prefix + key + '.');
      }
      else {
        into[prefix + key] = val;
      }
    }
  });

  return into;
};
```

It recursively extracts all public methods of the daterange object and stores them as "daterange.publicmethod" referances. This cause the original type key dateRange be undefined when `validateAttr` is called. When adding a stupid check (`key === "dateRange"`) as a last case to the instanceof blacklist, forcing it to resolve to the else block (`into[prefix + key] = val`), the validation work as expected.

Any ways to force the flatten method to skip the recursive extract and resolve to the else block?
*Using backbone-validation-amd.js v0.9.1*

**Comment by platinumazure** [comment]
Hi @kentandersen, I think pull request #264 should fix this issue (and that issue #261 is probably a duplicate of this one, oops). Please take a look and see if you agree.

**Comment by kentandersen** [comment]
Yes the #261 addresses the same problem I've had with validating nested objects, as do #186. The solution outlined in https://github.com/thedersen/backbone.validation/pull/264#issuecomment-71581538 seams like a good way fix this issiue

**Comment by platinumazure** [comment]
This should now be fixed with v0.11.1. Please let us know if you have any problems using that version or a later one.

@chiefGui do you want to close this?

**Comment by chiefGui** [comment]
Thank you @platinumazure.
